index.js

```diff
@@ -29,67 +28,64 @@ DatabaseUpdates.prototype = Object.create(EventEmitter.prototype)
-      .filter(i => i && semver.valid(i.split('-')[0]))
+      .filter((i) => i && semver.valid(i.split('-')[0]))
       // exclude anything after a hyphen from the version number
       .sort((a, b) => semver.compare(a.split('-')[0], b.split('-')[0]))
   } catch (e) {
     this.logger.error('Error reading updatePath folder', e)
   }
 }
 
-DatabaseUpdates.prototype.updateExists = function updateExists(file, cb) {
-  this.updateCollection.count({ file }, (err, count) => {
-    if (err) return cb(err)
-    return cb(null, count !== 0)
-  })
+DatabaseUpdates.prototype.updateExists = function updateExists(file) {
+  return this.updateCollection.count({ file }).then((count) => count !== 0)
 }
 
-DatabaseUpdates.prototype.runFile = function runFile(file, next) {
-  this.updateExists(file, (err, exists) => {
-    if (err) return next(err)
-    if (exists) return next()
-
-    this.emit('file', file)
-    this.logger.info('Running update:', file)
+DatabaseUpdates.prototype.runFile = async function runFile(file) {
+  const exists = await this.updateExists(file)
+  if (exists) return Promise.resolve()
 
-    /* eslint-disable global-require, import/no-dynamic-require */
-    const update = require(path.join(this.updatePath, file))
-    /* eslint-enable global-require, import/no-dynamic-require */
+  this.emit('file', file)
+  this.logger.info('Running update:', file)
 
-    return update(this.db, (err) => {
-      if (err) {
-        this.logger.error('Error running update:', file)
-        return next(err)
-      }
+  /* eslint-disable global-require, import/no-dynamic-require */
+  const update = require(path.join(this.updatePath, file))
+  /* eslint-enable global-require, import/no-dynamic-require */
 
-      return this.persistUpdate(file, next)
+  return update(this.db)
+    .then(() => this.persistUpdate(file))
+    .catch((err) => {
+      this.logger.error('Error running update:', file)
+      throw err
     })
-  })
 }
 
-DatabaseUpdates.prototype.persistUpdate = function persistUpdate(file, next) {
+DatabaseUpdates.prototype.persistUpdate = function persistUpdate(file) {
   this.logger.info('Persisting update:', file)
 
   const update = { file, created: new Date() }
-  this.updateCollection.insert(update, (err) => {
-    if (err) {
-      this.logger.error('Error persisting update:', update)
-      return next(err)
-    }
-
-    return next()
+  return this.updateCollection.insertOne(update).catch((err) => {
+    this.logger.error('Error persisting update:', update)
+    throw err
   })
 }
 
-DatabaseUpdates.prototype.run = function run() {
-  async.eachSeries(this.updateFiles, this.runFile.bind(this), (err) => {
-    if (err) return this.logger.error('Error running updates:', err)
+DatabaseUpdates.prototype.run = async function run() {
+  try {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const file of this.updateFiles) {
+      // eslint-disable-next-line no-await-in-loop
```

Comment:
don't want to build an array and use `Promise.all` or `Promise.allSettled`? heh

Reply:
Promise.all executes everything at once, whereas we want to apply the files sequentially in the order we find them.

Reply:
oh, yeah. I forgot that. good call then.